frame/support/src/signed_extensions.rs

```diff
@@ -0,0 +1,327 @@
+	fn eq(&self, other: &Self) -> bool {
+		self.ext == other.ext && self.adjuster == other.adjuster
+	}
+}
+
+impl<S: Eq, M, const V: TransactionPriority> Eq for AdjustPriority<S, M, V> {}
+
+impl<S: Encode, M, const V: TransactionPriority> Encode for AdjustPriority<S, M, V> {
+	fn encode(&self) -> Vec<u8> {
+		self.ext.encode()
+	}
+	fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+		self.ext.using_encoded(f)
+	}
+
+	fn size_hint(&self) -> usize {
+		self.ext.size_hint()
+	}
+
+	fn encode_to<T: codec::Output + ?Sized>(&self, dest: &mut T) {
+		self.ext.encode_to(dest)
+	}
+
+	fn encoded_size(&self) -> usize {
+		self.ext.encoded_size()
+	}
+}
+
+impl<S: Decode, M, const V: TransactionPriority> Decode for AdjustPriority<S, M, V> {
+	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
+		Ok(Self { ext: S::decode(input)?, adjuster: Default::default() })
+	}
+
+	fn encoded_fixed_size() -> Option<usize> {
+		S::encoded_fixed_size()
+	}
+}
+
+impl<S, M, const V: TransactionPriority> SignedExtension for AdjustPriority<S, M, V>
+where
+	S: SignedExtension,
+	M: Mode + Send + Sync,
+{
+	type AccountId = S::AccountId;
+	type AdditionalSigned = S::AdditionalSigned;
+	type Call = S::Call;
+	type Pre = S::Pre;
+
+	const IDENTIFIER: &'static str = S::IDENTIFIER;
+
+	fn additional_signed(&self) -> Result<Self::AdditionalSigned, TransactionValidityError> {
+		self.ext.additional_signed()
+	}
+
+	fn validate(
+		&self,
+		who: &Self::AccountId,
+		call: &Self::Call,
+		info: &DispatchInfoOf<Self::Call>,
+		len: usize,
+	) -> TransactionValidity {
+		let mut validity = self.ext.validate(who, call, info, len)?;
+		validity.priority = M::combine(validity.priority, V);
+		Ok(validity)
+	}
+
+	fn pre_dispatch(
+		self,
+		who: &Self::AccountId,
+		call: &Self::Call,
+		info: &DispatchInfoOf<Self::Call>,
+		len: usize,
+	) -> Result<Self::Pre, TransactionValidityError> {
+		self.ext.pre_dispatch(who, call, info, len)
+	}
+
+	fn validate_unsigned(
+		call: &Self::Call,
+		info: &DispatchInfoOf<Self::Call>,
+		len: usize,
+	) -> TransactionValidity {
+		let mut validity = S::validate_unsigned(call, info, len)?;
+		validity.priority = M::combine(validity.priority, V);
+		Ok(validity)
+	}
+
+	fn pre_dispatch_unsigned(
+		call: &Self::Call,
+		info: &DispatchInfoOf<Self::Call>,
+		len: usize,
+	) -> Result<Self::Pre, TransactionValidityError> {
+		S::pre_dispatch_unsigned(call, info, len)
+	}
+
+	fn post_dispatch(
+		pre: Self::Pre,
+		info: &DispatchInfoOf<Self::Call>,
+		post_info: &PostDispatchInfoOf<Self::Call>,
+		len: usize,
+		result: &DispatchResult,
+	) -> Result<(), TransactionValidityError> {
+		S::post_dispatch(pre, info, post_info, len, result)
+	}
+
+	fn identifier() -> Vec<&'static str> {
+		S::identifier()
+	}
+}
+
+/// Combination mode for the adjuster.
+pub trait Mode {
+	/// Return a combination of two transaction priorities.
+	fn combine(a: TransactionPriority, b: TransactionPriority) -> TransactionPriority;
```

Comment:
```suggestion
	fn combine_priority(a: TransactionPriority, b: TransactionPriority) -> TransactionPriority;
```

Because there could be more of this in the future?